### DISCUSSION TITLE
Watermark issue

### DISCUSSION THREAD

**Discussion by marrtn** [sumatrapdfreader/sumatrapdf#3279] [General]
Hello,

I have some specification documents with watermarks that ae shown in solid black in SumatraPDF 3.4.6, instead of some transparent grey (in Adobe Reader). Unfortunately, I am not allowed to share these documents.
Below an example:

![grafik](https://user-images.githubusercontent.com/18465098/219589280-5c1f6cf2-dada-44a5-85e6-3ac3a876bc74.png)
![grafik](https://user-images.githubusercontent.com/18465098/219589352-9ce89741-74af-47cd-a7b3-f59482ece7db.png)

Whenever there is some background color (the cell with shadow), the watermark seems to be correct, but without background it is solid black making the text below very hard to read.

Is there something I can do about that?
For raising an issue ticket, I'd need some example-PDF which I can't provide right now.

There are some similar discussions/issue to be found, but I couldn't find a solution.

Regards
 Martin

**Comment by GitHubRulesOK** [comment]
Such watermarks are often odded into the existing document after receipt/generation, however since that is (NOT SO) clearly underlay'd it must have been generated by the PDF producer, without sample cant say, but its very easy to remove them so there is no problem with convert to archive format etc., and if desired replace with a different underlay method that could be a whiter image so no interference with the contents when reading aloud (accessibility requires non standard translucent objects be removed) for screen read/printing etc.

Ensure your testing with current pre-release, since only that can be altered if it has not already been fixed ?

**Comment by kjk** [comment]
See if it still happens in pre-release version: https://www.sumatrapdfreader.org/prerelease

**Comment by daysneco** [comment]
Hi did you solve this issue, i also have this problem

**Comment by marrtn** [comment]
Sorry for my late reply...
The current pre-Release fixes the issue for me. Thanks a lot!

**Comment by brianrho** [comment]
Hi, I'm using Sumatra 3.5.2 (released in October last year) but I still have this issue, the watermark completely obscures any overlapping text. Did the pre-release fix ever go out?

**Reply by GitHubRulesOK** [reply]
@brianrho  
you will need to test most recent e.g. currently August https://www.sumatrapdfreader.org/prerelease 
portable should be good enough without re-installing

**Reply by brianrho** [reply]
Thanks, I dont mind testing with that, but is there a plan to make this fix available as part of the usual releases, so I can eventually return to that? 

(I'd assumed that any fixes in the May 2023 prerelease would already be in the later October release.)

**Reply by GitHubRulesOK** [reply]
@brianrho 

There is never a timeframe for a release it may be today or a few years from now. (e.g big time gap from 2016-08 to 2020-03 with lots of pre-releases)

Pre-Release is usually just as stable (and with security fixes) as the release with glitches like watermark problems 

The main difference is any known instability or experimental function will be removed from formal official 3.6 release.

**Reply by brianrho** [reply]
Right, so was this specific fix deemed too unstable / experimental to be cherry-picked for the October release? Or was it simply not considered?

If the latter, it would be nice to include it in the next minor release whenever that happens.

**Reply by GitHubRulesOK** [reply]
@brianrho 
the Feb 2023 problem above and several others were fixed prior to May 2023 (https://github.com/sumatrapdfreader/sumatrapdf/issues/2934) you clearly must have a different one if the 3.5.2 (2023-10) is still showing an issue.

Several were fixed in more recent MuPDF updates, hence the suggestion to always try latest pre-release or supply a file for testing with a view to fixing if current MuPDF render still has an issue.

**Reply by brianrho** [reply]
Sorry, this is coming after a while -- I tried the pre-release then (3.16.16779) like you suggested, but the issue is still present.

Unfortunately, I can't provide a sample file since all the files I have with this watermark, are confidential, not mine to release. Any other suggestions?

**Reply by GitHubRulesOK** [reply]
@brianrho 
analyse if it is a MuPDF issue
Currently am running an Easter version so 1.26 but **1.25.2** (latest WindowsX64 ZIP) [MuPDF-GL](https://www.mupdf.com/releases) is similar to SumatraPDF core code as based on MuPDF thus if there is a problem there it is Artifex (PDF.co) need to fix their code but they WILL require a true sample or wont attempt to provide a fix. Unknown when they will release a Windows 1.26 but should be soon due to lots of activity so worth keeping an eye out for that and test if it solves problems since many different changes over recent months

**Reply by brianrho** [reply]
Thanks -- so if I understand correctly:

- the latest Sumatra release is based on MuPDF 1.25.2, which is the latest _Windows_ MuPDF release
- there are more recent MuPDF releases e.g. 1.2.6, containing many fixes but we have to wait till they make a Windows version of that release, which should hopefully be soon

In that case -- is there a way to check which MuPDF version I have in my Sumatra installation? And what's the recommended way to update the MuPDF used by my Sumatra installation?

**Reply by GitHubRulesOK** [reply]
@brianrho 
Not really, SumatraPDF uses parts of MuPDF thus not always producing exactly same outputs but if a pdf is a problem in MuPDF or a problem fixed there then the code improvements usually  filter through or KJK asks for the problem to be fixed but as SumatraPDF is a "variation" the requests to fix are variable in response from hours to years or more :-) or even MuPDF may say they "WontFix" as not in their purview!

The most recent update of SumatraPDF  embedded MuPDF code was `update mupdf to 1.25-rc2  =  5 months ago` thus not even the full 1.25 nor more recent (so due for a large overhaul,) but have absolutely no idea if/when that may happen, as KJK seems to be busy elsewhere.

**Reply by brianrho** [reply]
Okay, I guess I'll just have to wait and hope that (1) MuPDF 1.2.6 holds the solution to my problem (2) a Windows release is made soon so that (3) you guys can also merge it into a SumatraPDF pre-release soon after.

**Comment by detuur** [comment]
I'm having a similar issue. I can't share the PDF itself as it's a datasheet under NDA, but I can perhaps give you guys a few leads.

The PDF producer field is:

>Microsoft® Word for Microsoft 365; modified using iTextSharp™ 5.5.13.3 ©2000-2022 iText Group NV (AGPL-version)

When inspecting it in a PDF editor (PDFGear), it seems like the watermark is simply a black-and-white image added individually to every page.

I've opened it in RUPS to inspect the structure, and this is what a page looks like: 
<img width="418" height="773" alt="image" src="https://github.com/user-attachments/assets/7cbbdafd-6ba6-4daa-84fd-98fdd1329c84" />

It seems likely to me that the `/Xi3` object in the `/ExtGState` dict is the image mode of the `/Xi5` `/XObject`, which is the watermark image object. That would mean that it is achieving the transparency effect by using the **multiply** image mode. This seems to track, as Sumatra renders the transparency correctly where the watermark intersects with other images in the PDF, but somehow breaks over empty sections and plain text. Sounds like the multiply only works correctly when there is other image data to multiply?

None of this is a professional analysis, as I don't know much of how PDF works, I'm not familiar with the spec, so I'm just giving my best guess. Hope it helps though.

**Reply by GitHubRulesOK** [reply]
@detuur whilst a RUPS view has some hints of construction the way a PDF accumulates graphic states is not shown in a meaningful way so even emulating some values may not show the combined state hence the request to always provide the Physical file. There are several factors and permutations such as Word iText Adobe (wooly standards) and then versions of MuPDF and its fixes.

1) If not a problem in current pre-release then nothing to fix (no retro fix possible).
2) For a fix the developer KJK needs the physical file (or a true working ex-sample).

**Comment by GitHubRulesOK** [comment]
**For others landing here** 

Always test in pre-release and if the problem is still present  then 
PLEASE WITH A SAMPLE open an issue.

@kjk the OP issue was resolved so I suggest this thread is fit to close ? 
https://github.com/sumatrapdfreader/sumatrapdf/discussions/3279#discussioncomment-5789313
> Sorry for my late reply... The current pre-Release fixes the issue for me. Thanks a lot!
